src/components/Navbar.jsx

```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useTheme } from '../context/ThemeContext';
+
+const Navbar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const router = useRouter();
+  const { theme, toggleTheme } = useTheme();
+
+  const toggleMenu = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const getLinkClass = (path) =>
+    `text-${theme === 'dark' ? 'white' : 'black'} hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium ${
```

Comment:
**Issue**: The use of template literals with dynamic values inside className can lead to CSS injection vulnerabilities if not properly sanitized. <br> **Fix**: Ensure that any dynamic values used within template literals for className are sanitized to prevent CSS injection attacks. Consider using a dedicated library for handling class names that ensures safe binding of dynamic values. <br> **Code Suggestion**: 
 ```
 - 'text-${theme === 'dark' ? 'white' : 'black'} hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium ${
 + 'text-${theme === 'dark' ? 'white' : 'black'} hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium ${sanitizeClassName(
 +   router.pathname === path ? 'font-bold' : ''
 + )} transition-colors duration-300';
 ```